src/main/java/org/folio/service/PatronBlocksService.java

```diff
@@ -71,20 +74,32 @@ public PatronBlocksService(Map<String, String> okapiHeaders, Vertx vertx) {
     }
 
     List<Future<AutomatedPatronBlock>> futures = limits.stream()
-      .filter(limit -> isConditionLimitExceeded(summary, limit))
-      .map(this::createBlockForLimit)
+      .map(limit -> createBlockForLimit(userSummary, limit))
       .collect(Collectors.toList());
 
     return CompositeFuture.all(new ArrayList<>(futures))
       .onFailure(log::error)
-      .onSuccess(cf -> blocks.getAutomatedPatronBlocks().addAll(cf.list()))
+      .onSuccess(cf -> blocks.getAutomatedPatronBlocks().addAll(
+        Arrays.stream(Condition.values())
+          .map(condition -> cf.list().stream()
+            .map(AutomatedPatronBlock.class::cast)
+            .filter(Objects::nonNull)
+            .filter(e -> e.getPatronBlockConditionId().equals(condition.getId()))
```

Comment:
Please think of a better name for a variable

Reply:
This piece was removed